### PR TITLE
certificate updates from feedback

### PR DESCRIPTION
**Pull request by slkendell-mt**
- Refactored exports controller & introduced new service. 
- Added account holder for new reg certs
- Added missing fields
- Changes and UI tweaks as per feedback

### CODE REVIEW

File: service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    this.registrationService = registrationService;
+    this.legacyBeaconService = legacyBeaconService;
+    this.exportMapper = exportMapper;
+    this.noteService = ns;
+    this.accountHolderService = accountHolderService;
+  }
+
+  public BeaconExportDTO getBeaconExportDTO(
+    UUID rawBeaconId,
+    String noteGeneratedType
```
Comment (Evie-Skinner18) [inline service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java:50]: So we're passing in a noteGeneratedType and that becomes part of the text for the note. Nice

File: service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  public LabelDTO getLabelDTO(UUID rawBeaconId) {
+    BeaconId beaconId = new BeaconId(rawBeaconId);
+
+    try {
+      Registration registration = registrationService.getByBeaconId(beaconId);
+      LabelDTO data = exportMapper.toLabelDTO(registration);
+
+      //Only create note for modern for now.
```
Comment (Evie-Skinner18) [inline service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java:92]: Just a thought, putting this step in a method called something like createNoteForModernBeacon() would remove the need for a comment

File: service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    UUID rawBeaconId,
+    String noteGeneratedType
+  ) {
+    BeaconId beaconId = new BeaconId(rawBeaconId);
+
+    try {
+      Registration registration = registrationService.getByBeaconId(beaconId);
+      AccountHolder accountHolder = accountHolderService
+        .getAccountHolder(registration.getBeacon().getAccountHolderId())
+        .orElse(null);
```
Comment (Evie-Skinner18) [inline service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java:58]: now this code is in its own service, we could unit test it. Perhaps one for a future iteration though

File: service/src/main/java/uk/gov/mca/beacons/api/export/rest/BeaconExportAccountHolderDTO.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Valid
+  private String fullName;
+
+  @JsonUnwrapped
+  private AddressDTO address;
+
+  @Valid
+  // format: telephone   alternativeTelephoneNumber
+  private String telephoneNumbers;
```
Comment (Evie-Skinner18) [inline service/src/main/java/uk/gov/mca/beacons/api/export/rest/BeaconExportAccountHolderDTO.java:23]: Consider removing the comment and putting that meaning into the property name

**Review by Evie-Skinner18 — APPROVED** [review]

File: backoffice/src/views/exports/certificates/Certificate.tsx [hunk 5]
```diff
@@ -96,7 +101,7 @@ const BeaconSection = ({ beacon }: BeaconExportProps): JSX.Element => {
         value={customDateStringFormat(beacon.batteryExpiryDate, "MMMM yyyy")}
       />
       <CertificateField
-        classes="half"
+        classes="full"
```
[the remark this answers is not in the document]
Comment (slkendell-mt) [reply backoffice/src/views/exports/certificates/Certificate.tsx:104]: Yeah, half is 50%, full is 100% 

So

---half-----  ----half----
---------full------------

File: service/src/main/java/uk/gov/mca/beacons/api/beaconuse/domain/BeaconUse.java [hunk 6]
```diff
@@ -296,4 +298,29 @@ public String getName() {
 
     return Objects.requireNonNullElse(name, "");
   }
+
+  public List<String> getCommunicationTypes() {
+    List<String> communicationTypes = new ArrayList<>();
+
+    if (BooleanUtils.isTrue(vhfRadio)) {
+      communicationTypes.add("VHF Radio");
```
[the remark this answers is not in the document]
Comment (slkendell-mt) [reply service/src/main/java/uk/gov/mca/beacons/api/beaconuse/domain/BeaconUse.java:306]: Yeah I agree with all of this, was thinking about some sort of Map of bool->String Name so you can just add another one in that. 

Enums are a good shout too, but was looking for a quick way to do it for now, with the aim of going back and improving. 

Unit tests are a very good point, I did mean to go back and add some. Will do so now.

I think for a first pass keep it as is, with the aim to migrate to enums in the future

File: service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java (hunk 2 above)
Comment (slkendell-mt) [reply service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java:92]: Agreed, ideally legacy would also have notes .

File: service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java (hunk 3 above)
Comment (slkendell-mt) [reply service/src/main/java/uk/gov/mca/beacons/api/export/application/ExportService.java:58]: Yeah there's nothing really checking what happens if a beacon doesn't have an account holder. one for a test I agree

**Pull request merged**